src/Badges.sol

```diff
@@ -148,9 +156,21 @@ contract Badges is
       "giveToMany: unauthorized"
     );
 
+    address[] memory invalidRecipients = new address[](_recipients.length);
```

Comment:
I had to set it to _some_ specified length here since you only do dynamically sized arrays with storage variables